core-bundle/src/Controller/ContentElement/MarkdownController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\ContentElement;
+
+use Contao\Config;
+use Contao\ContentModel;
+use Contao\FilesModel;
+use Contao\Template;
+use League\CommonMark\Environment;
+use League\CommonMark\Extension\Autolink\AutolinkExtension;
+use League\CommonMark\Extension\ExternalLink\ExternalLinkExtension;
+use League\CommonMark\Extension\Strikethrough\StrikethroughExtension;
+use League\CommonMark\Extension\Table\TableExtension;
+use League\CommonMark\Extension\TaskList\TaskListExtension;
+use League\CommonMark\MarkdownConverter;
+use League\CommonMark\MarkdownConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+
+class MarkdownController extends AbstractContentElementController
+{
+    /**
+     * @var MarkdownConverterInterface
+     */
+    private $converter;
+
+    public function __construct(MarkdownConverterInterface $converter)
+    {
+        $this->converter = $converter;
+    }
+
+    public static function createController(RequestStack $requestStack): self
```

Comment:
This factory method looks wrong to me. I can imagine a lot of these options being runtime configuration for the content element (e.g. whether to generate `target=_blank`), I feel like this should go into the response method (or be a private/protected method that's being used at runtime).

Reply:
That would mean we would create the instance of the MarkdownConverter on the fly rather than injecting it as a dependency. So in the `getResponse()` we'd have `new MarkdownConverter()` which felt a bit wrong to me.